app/gui/view/graph-editor/src/component/edge.rs

```diff
@@ -4,2033 +4,951 @@ use crate::prelude::*;
-                let focus_split_angle    = focus_split_angle.into();
-                let focus_split_center   = focus_split_center.px();
-
-                let shape_padding = -1.px();
-                let shape         = Triangle(width+&shape_padding,height+&shape_padding);
-                let shape         = FocusedEdge::new(shape,&focus_split_center,&focus_split_angle);
-                let shape         = shape.fill(&color, &focus_color);
-                shape.into()
-            }
-        }
+impl Edge {
+    /// Constructor.
+    #[profile(Detail)]
+    pub fn new(app: &Application) -> Self {
+        let frp = Frp::new();
+        let model = Rc::new(EdgeModel::new(&app.display.default_scene));
 
-        impl EdgeShape for View {
-            fn set_focus_split_center_local(&self, center:Vector2<f32>) {
-                // We don't want the arrow to be half-focused. The focus split point is set to the
-                // closest edge (all or nothing).
-                let min = -Vector2(ARROW_SIZE_X,ARROW_SIZE_Y);
-                let max =  Vector2(ARROW_SIZE_X,ARROW_SIZE_Y);
-                let mid =  Vector2::<f32>::zero();
-                let x   = if center.x < mid.x { min.x } else { max.x };
-                let y   = if center.y < mid.y { min.y } else { max.y };
-                self.focus_split_center.set(Vector2(x,y));
-            }
+        let network = &frp.network;
 
-            fn set_focus_split_angle(&self, angle:f32) {
-                 self.focus_split_angle.set(angle);
-            }
+        let edge_color = color::Animation::new(network);
 
-            fn events(&self) -> &PointerTarget_DEPRECATED {
-                &self.events_deprecated
-            }
+        use ensogl::control::io::mouse;
+        let display_object = &model.display_object;
+        let mouse_move = display_object.on_event::<mouse::Move>();
+        let mouse_down = display_object.on_event::<mouse::Down>();
+        let mouse_out = display_object.on_event::<mouse::Out>();
 
-            fn set_color(&self, color:color::Rgba) {
-                self.color_rgba.set(Vector4(color.red,color.green,color.blue,color.alpha));
-            }
+        let output = &frp.private.output;
+        frp::extend! { network
+            // Setters. Changes to these values don't automatically trigger a redraw.
+            eval frp.target_position ((t) model.set_target_position(*t));
+            eval frp.source_attached ((t) model.set_source_attached(*t));
+            eval frp.target_attached ((t) model.set_target_attached(*t));
+            eval frp.source_width    ((t) model.set_source_width(*t));
+            eval frp.source_height   ((t) model.set_source_height(*t));
+            eval frp.set_disabled    ((t) model.set_disabled(*t));
+
+            // Mouse events.
+            eval mouse_move            ((e) model.set_mouse_position_and_redraw(e.client_centered()));
```

Comment:
We no longer use spacing in FRP code, and it isn't consistent here anyway, so I suggest removing redundant spaces.